graphql/course/mutations.ts

```diff
@@ -0,0 +1,144 @@
+import { prisma } from "../../common/prisma";
+import { Resolver, Mutation, Root, Arg, Authorized, Ctx } from "type-graphql";
+import * as GraphQLModel from "../generated/models";
+import { AuthorizedDeferred, hasAccess, Role } from "../authorizations";
+import * as TypeGraphQL from "type-graphql";
+import { course_category_enum } from "@prisma/client";
+import { getLogger } from "log4js";
+import { getSessionStudent } from "../authentication";
+import { GraphQLContext } from "../context";
+import { getCourse, getStudent } from "../util";
+
+class PublicCourseCreateInput {
+  @TypeGraphQL.Field(_type => String, {
+      nullable: false
+  })
+  name!: string;
+
+  @TypeGraphQL.Field(_type => String, {
+      nullable: false
+  })
+  outline!: string;
+
+  @TypeGraphQL.Field(_type => String, {
+      nullable: false
+  })
+  description!: string;
+
+  // @TypeGraphQL.Field(_type => String, {
+  //   nullable: true
+  // })
+  // image?: string | undefined;
+
+  @TypeGraphQL.Field(_type => course_category_enum, {
+      nullable: false
+  })
+  category!: "revision" | "club" | "coaching";
+
+  @TypeGraphQL.Field(_type => Boolean, {
+      nullable: true
+  })
+  allowContact?: boolean | undefined;
```

Comment:
this could be non-nullable. Also the other `nullable: false` are obsolete, they're the default value ... Maybe this could be adressed in a followup change